src/lib/operators/export_binary.cpp

```diff
@@ -255,17 +255,130 @@ void ExportBinary::_write_segment(const RunLengthSegment<T>& run_length_segment,
   export_values(ofstream, *run_length_segment.end_positions());
 }
 
-void ExportBinary::_export_attribute_vector(std::ofstream& ofstream, const CompressedVectorType type,
-                                            const BaseCompressedVector& attribute_vector) {
+template <typename T>
+void ExportBinary::_write_segment(const FrameOfReferenceSegment<T>& frame_of_reference_segment,
+                                  std::ofstream& ofstream) {
+  Fail("Frame of Reference Segments not implemented for data type");
+}
+
+template <>
+void ExportBinary::_write_segment(const FrameOfReferenceSegment<int32_t>& frame_of_reference_segment,
+                                  std::ofstream& ofstream) {
+  export_value(ofstream, EncodingType::FrameOfReference);
+
+  const auto offset_value_vector_width = [&]() {
+    Assert(frame_of_reference_segment.compressed_vector_type(),
+           "Expected FrameOfReferenceSegment to use vector compression for offset value vector");
+    switch (*frame_of_reference_segment.compressed_vector_type()) {
+      case CompressedVectorType::FixedSize4ByteAligned:
+        return 4u;
+      case CompressedVectorType::FixedSize2ByteAligned:
+        return 2u;
+      case CompressedVectorType::FixedSize1ByteAligned:
+        return 1u;
+      default:
+        Fail("Any other type should have been caught before.");
+    }
+  }();
+
+  // Write attribute vector width
+  export_value(ofstream, static_cast<AttributeVectorWidth>(offset_value_vector_width));
```

Comment:
Shouldn't `_export_compressed_vector` handle this? That way, we wouldn't have to duplicate the above part, would have better separation, and would have an easier time implementing additional compression types.